packages/db-schema/src/repositories/types/index.ts

```diff
@@ -0,0 +1,172 @@
+import { z } from "zod/v4";
+
+import {
+  actorStatus,
+  competitionAgentStatus,
+  competitionStatus,
+  competitionType,
+} from "../../core/defs.js";
+
+/**
+ * Zod schema for the status of a user, agent, or admin.
+ */
+export const ActorStatusSchema = z.enum(actorStatus.enumValues);
+
+/**
+ * Status of a user, agent, or admin.
+ */
+export type ActorStatus = z.infer<typeof ActorStatusSchema>;
+
+// Zod schema for SpecificChain validation
+export const SpecificChainSchema = z.enum([
+  "eth",
+  "polygon",
+  "bsc",
+  "arbitrum",
+  "optimism",
+  "avalanche",
+  "base",
+  "linea",
+  "zksync",
+  "scroll",
+  "mantle",
+  "svm",
+]);
+
+// Type derived from the Zod schema
+export type SpecificChain = z.infer<typeof SpecificChainSchema>;
+
+/**
+ * Zod schema for the status of a competition.
+ */
+export const CompetitionStatusSchema = z.enum(competitionStatus.enumValues);
```

Comment:
Some of the types in this file, such as this one, I don't see a corresponding change where this type is being moved from somewhere else.  In fact, looking at the competition controller, it seems to be still be importing a same-named type from a different location.  I think we probably don't want to introduce duplicate types here and instead want to migrate the existing calls to use these new versions?

Reply:
Correct, I just copied the types over as a short term solution for the sake of simplicity. These types are still defined and used in the api app. The way types are defined and used in `api` is pretty confusing, so I didn't want to put a lot more work into preserving that or re-inventing it. I figured we can come up with a good approach for types when we implement the new api and shared services.